### PR TITLE
Adding the sponsor to the lede banner

### PR DESCRIPTION
**Pull request by itsjoekent**
### What does this PR do?
- Adding the sponsor to the lede banner
- Shifting the arrow logic around a bit. In the next PR we can deal with the right angle arrow.

#### Large
screen shot 2018-01-08 at 1 54 26 pm

#### Medium
[see updated photo in comment](https://github.com/DoSomething/phoenix-next/pull/593#discussion_r160721634)

#### Small
screen shot 2018-01-08 at 1 53 58 pm

### What are the relevant tickets/cards?
https://www.pivotaltracker.com/story/show/154037769

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
     "heroku-postbuild": "sh bootstrap/setup.sh"
   },
   "engines": {
-    "node": "6.x",
-    "npm": "4.x"
+    "node": "8.x",
+    "npm": "5.x"
```
Comment (weerd) [inline package.json]: 🎉

File: resources/assets/components/CampaignSignupArrow/CampaignSignupArrow.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import React from 'react';
 import './campaignSignupArrow.scss';
 
 const CampaignSignupArrow = ({ content, className }) => (
```
Comment (weerd) [inline resources/assets/components/CampaignSignupArrow/CampaignSignupArrow.js]: Not something I caught before but could we rename this component to `MessageCallout`? It's more than just an arrow 😬  Not necessary for this PR, but maybe do it in a quick follow-up PR?

File: resources/assets/components/LedeBanner/lede-banner.scss [hunk 3]
```diff
@@ -16,13 +16,17 @@
     width: 100%;
     height: 70px;
 
-    @include media($medium) {
+    @include media($large) {
```
Comment (weerd) [inline resources/assets/components/LedeBanner/lede-banner.scss]: @lkpttn maybe this is more of a question for you... if we wait till Large size to switch to the 50/50 in the lede banner, the banner will take up a lot of vertical space. Could we just keep the 50/50 split on Medium size, but the join button, message callout and sponsor stay stacked vertically until Large? cc: @itsjoekent

File: resources/assets/components/CampaignSignupArrow/CampaignSignupArrow.js (hunk 2 above)
Comment (itsjoekent) [reply resources/assets/components/CampaignSignupArrow/CampaignSignupArrow.js]: sure

File: package.json (hunk 1 above)
Comment (mendelB) [reply package.json]: image
🎉

File: resources/assets/components/LedeBanner/lede-banner.scss (hunk 3 above)
Comment (lkpttn) [reply resources/assets/components/LedeBanner/lede-banner.scss]: I just talked to Diego about this, he had a good suggestion about using 50% of the space in the banner at the medium size, but keeping the half/half layout. He can explain it more but I approved it!

Comment (weerd) [reply resources/assets/components/LedeBanner/lede-banner.scss]: @itsjoekent lets chat tomorrow about it!

File: package.json (hunk 1 above)
Comment (mendelB) [reply package.json]: image

File: resources/assets/components/LedeBanner/lede-banner.scss (hunk 3 above)
Comment (itsjoekent) [reply resources/assets/components/LedeBanner/lede-banner.scss]: screen shot 2018-01-10 at 11 00 44 am

here is the actual screenshot i should have uploaded instead for medium

File: package.json (hunk 1 above)
Comment (weerd) [reply package.json]: escalated

File: resources/assets/components/LedeBanner/lede-banner.scss [hunk 4]
```diff
@@ -16,13 +16,17 @@
...
       width: 50%;
     }
   }
 
   .header__signup {
     padding-top: $base-spacing;
+
+    .button.-float {
```
Comment (weerd) [inline resources/assets/components/LedeBanner/lede-banner.scss]: Probably don't need the modifier class when already targeting the `.button` from within the `.header__signup`. Should just be fine w/out `.-float`.

Comment (itsjoekent) [reply resources/assets/components/LedeBanner/lede-banner.scss]: the -float is intentional, if we apply the float when there isnt stuff under the button then the padding breaks and the button aligns with the bottom of the lede banner

File: resources/assets/components/LedeBanner/templates/MosaicTemplate.js [hunk 5]
```diff
@@ -25,14 +28,32 @@ const MosaicTemplate = (props) => {
   };
 
   const signupArrowComponent = signupArrowContent ? (
-    <CampaignSignupArrow content={signupArrowContent} className="-mosaic-arrow" />
+    <div className="lede-banner__arrow">
```
Comment (weerd) [inline resources/assets/components/LedeBanner/templates/MosaicTemplate.js]: Just curious, is this extra `div` and class necessary? Couldn't you just apply the styles to the `message-callout`?

File: resources/assets/components/LedeBanner/lede-banner.scss (hunk 4 above)
Comment (weerd) [reply resources/assets/components/LedeBanner/lede-banner.scss]: Ah ok!

File: resources/assets/components/LedeBanner/templates/MosaicTemplate.js (hunk 5 above)
Comment (itsjoekent) [reply resources/assets/components/LedeBanner/templates/MosaicTemplate.js]: it doesnt seem to make any visual difference when i remove the div and apply the classname

File: resources/assets/components/LedeBanner/templates/MosaicTemplate.js [hunk 6]
```diff
@@ -25,14 +28,30 @@ const MosaicTemplate = (props) => {
   };
 
   const signupArrowComponent = signupArrowContent ? (
-    <CampaignSignupArrow content={signupArrowContent} className="-mosaic-arrow" />
+    <CampaignSignupArrow content={signupArrowContent} className="lede-banner__arrow -mosaic-arrow" />
```
Comment (weerd) [inline resources/assets/components/LedeBanner/templates/MosaicTemplate.js]: Also meant that without that extra `div`, you likely don't need the `lede-banner__arrow` class either and can just shift those styles over to the `message-callout` class, right?

**Pull request merged**